src/modules/always_on_top/dllmain.cpp

```diff
@@ -0,0 +1,272 @@
+#include "pch.h"
+#include <interface/powertoy_module_interface.h>
+#include <interface/powertoy_system_menu.h>
+#include <common/settings_objects.h>
+#include <set>
+#include "trace.h"
+#include "resource.h"
+
+extern "C" IMAGE_DOS_HEADER __ImageBase;
+
+BOOL APIENTRY DllMain(HMODULE hModule, DWORD  ul_reason_for_call, LPVOID lpReserved) {
+  switch (ul_reason_for_call) {
+  case DLL_PROCESS_ATTACH:
+    Trace::RegisterProvider();
+    break;
+  case DLL_THREAD_ATTACH:
+  case DLL_THREAD_DETACH:
+    break;
+  case DLL_PROCESS_DETACH:
+    Trace::UnregisterProvider();
+    break;
+  }
+  return TRUE;
+}
+
+const static wchar_t* MODULE_NAME = L"Always On Top";
+const static wchar_t* MODULE_DESC = L"Adds a command to the title bar context menu to set a window on top of all others.";
+const static wchar_t* HOTKEY_NAME = L"AlwaysOnTop_HotKey";
+const static wchar_t* HOTKEY_WINDOW_CLASS_NAME = L"HotkeyHandleWindowClass";
+const static wchar_t* OVERVIEW_LINK = L"https://github.com/microsoft/PowerToys/blob/master/src/modules/always_on_top/README.md";
+
+struct AllwaysOnTopSettings {
+  // Default hotkey WIN + ALT + T
+  PowerToysSettings::HotkeyObject editorHotkey =
+    PowerToysSettings::HotkeyObject::from_settings(true, false, true, false, 0x54, L"T"); // ASCII code for T 0x54
+};
+
+class AlwaysOnTop : public PowertoyModuleIface {
+private:
+
+  bool                     enabled{ false };
+  HWND                     hotKeyHandleWindow{ nullptr };
+  HWND                     currentlyOnTop{ nullptr };
+  AllwaysOnTopSettings     settings;
+  std::wstring             itemName{ MODULE_NAME };
+  PowertoySystemMenuIface* systemMenuHelper{ nullptr };
+
+  void ProcessCommand(HWND window);
+  bool SetWindowOnTop(HWND window);
+  void ResetCurrentOnTop();
+  void CleanUp();
+
+  void LoadSettings(PCWSTR config, bool fromFile);
+  void SaveSettings();
+
+  std::vector<PowertoySystemMenuIface::ItemInfo> CustomItems();
+
+  LRESULT WndProc(HWND, UINT, WPARAM, LPARAM) noexcept;
+
+protected:
+
+  static LRESULT CALLBACK WndProc_Helper(HWND, UINT, WPARAM, LPARAM) noexcept;
+
+public:
+
+  AlwaysOnTop()
+  {
+    LoadSettings(MODULE_NAME, true);
+  }
+
+  virtual void destroy() override
+  {
+    CleanUp();
+    delete this;
+  }
+
+  virtual const wchar_t* get_name() override
+  {
+    return MODULE_NAME;
+  }
+
+  virtual const wchar_t** get_events() override
+  {
+    static const wchar_t* events[] = { nullptr };
+    return events;
+  }
+
+  virtual bool get_config(wchar_t* buffer, int* buffer_size) override
+  {
+    HINSTANCE hinstance = reinterpret_cast<HINSTANCE>(&__ImageBase);
+
+    PowerToysSettings::Settings common(hinstance, get_name());
```

Comment:
Change `common` to a better name, in other powertoys we call it `settings`, thanks.